### PR TITLE
Add ktlint plugin code styling and formatting

### PR DESCRIPTION
**Pull request by G10xy**
# Description

Based on the issue #279 I already used ktlint plugin, in order to mantain a standard about code style and formatting.
Please have check, give it a try. 
You can find in the Grandle task at the sub-task  formatting. 
By running ktlintFormat, the code will be automatically restyled and, in case of .* imports, it will generate a txt file reporting the error.

Some link about: 
https://reflectoring.io/code-format-with-ktlint/
https://www.baeldung.com/kotlin/ktlint-code-formatting

# Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Dependency update / replacement

# How Has This Been Tested?
I run ktlintFormat and checked that it reported all problems about importing and automatically formatting the code style.
I also tested it with gradle check and gradle build and it worked fine everytime. 

# Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation if necessary
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by andregasser** [comment]
Hello @G10xy 
Thanks a lot for this PR. As we are already using detekt in this project, I think it does not make much sense to add ktlint to the project as well. This is because detekt wraps ktlint formatting rules already:

image

Reference: https://detekt.dev/docs/intro

We are already using detekt formatting rules, which means, we're using ktlint already.

I think a way we could go is setting up an .editorconfig file in the root folder of our project. When the IDE starts, it reads .editorconfig and sets Kotlin/Java code formatting rules accordingly. One advantage I see when using .editorconfig over intelliJ .xml code formatting files is that other IDEs will most probably read this file as well.

Please let me know what you think and if you want to work on this. Otherwise, I'll pick this up.

Best regards,
André

Please let me know what you think of this.

File: bigbone/build.gradle [hunk 1]
```diff
@@ -1,6 +1,11 @@
 plugins {
     id 'bigbone.library-conventions'
     id 'bigbone.integrationtest-conventions'
+    id "org.jlleitschuh.gradle.ktlint" version "11.6.1"
```
Comment (andregasser) [inline bigbone/build.gradle:4]: This should go into the `buildSrc/src/main/groovy/bigbone.kotlin-conventions.gradle` file. Also, the version should be placed in the Gradle version catalog instead (`gradle/libs.versions.toml`).

File: bigbone/build.gradle [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
+}
+
+allprojects {
```
Comment (andregasser) [inline bigbone/build.gradle:7]: This is most probably no longer necessary, once above changes are implemented (kotlin conventions, version catalog).

**Comment by andregasser** [comment]
After doing a bit of research, I think we should keep `detekt` as a tool for code analysis, but use `ktlint` Gradle plugin for linting.
Especially after reading this: https://github.com/detekt/detekt/discussions/5997

I have added two minor comments to your PR. Would be great, if you could fix this. 
Also, we need to disable `detekt-formatting` entirely, which means, we need to remove the declarations of it from various places:
- Gradle version catalog file (`gradle/libs.versions.toml`)
- Gradle kotlin convention file (`buildSrc/src/main/groovy/bigbone.kotlin-conventions.gradle`), lines 7-9
- Remove the formatting rules from the detekt config file (`config/detekt/detekt.yml`)

File: bigbone/build.gradle (hunk 2 above)
Comment (G10xy) [reply bigbone/build.gradle:7]: I had to add _alias(libs.plugins.ktlint)_ in the build.gradle both of the bigbone and bigbone-rx and sample-kotlin to make the project build rightly

**Comment by G10xy** [comment]
> After doing a bit of research, I think we should keep `detekt` as a tool for code analysis, but use `ktlint` Gradle plugin for linting. Especially after reading this: [detekt/detekt#5997](https://github.com/detekt/detekt/discussions/5997)
> 
> I have added two minor comments to your PR. Would be great, if you could fix this. Also, we need to disable `detekt-formatting` entirely, which means, we need to remove the declarations of it from various places:
> 
>     * Gradle version catalog file (`gradle/libs.versions.toml`)
> 
>     * Gradle kotlin convention file (`buildSrc/src/main/groovy/bigbone.kotlin-conventions.gradle`), lines 7-9
> 
>     * Remove the formatting rules from the detekt config file (`config/detekt/detekt.yml`)
> 
> 
> detekt as a tool should remain in place, we just want to remove the detekt-formatting functionality.

Thanks, I honestly did not know so deeply about.

**Review by andregasser — APPROVED** [review]
Everything looks good! Thank you very much 👍

**Pull request merged**